annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -536,33 +587,117 @@ private TranscriptConsequenceSummary getCanonicalTranscript(VariantAnnotation gn
+                        gnResponseVariantKeyMap.put(extractGenomicLocationAsString(gnResponse.getAnnotationSummary().getGenomicLocation()), gnResponse);
+                    } else {
+                        LOG.warn("Annotation failed for variant " + gnResponse.getVariant());
+                    }
+                }
+            }
         }
-        // create annotated records by merging the responses from gn with their corresponding MAF record
+
         List<AnnotatedRecord> annotatedRecords = new ArrayList<>();
+        // loop through the original mutationRecords (in original sort order) and
+        // create annotated records by merging the responses from gn with their corresponding MAF record
         for (MutationRecord record : mutationRecords) {
             String genomicLocationKey = extractGenomicLocationAsString(record);
             AnnotatedRecord annotatedRecord = new AnnotatedRecord(record);
             // if not a failed annotation then convert/merge the response from gn with the maf record
             VariantAnnotation gnResponse = gnResponseVariantKeyMap.get(genomicLocationKey);
             if (gnResponse == null) {
-                summaryStatistics.addFailedAnnotatedRecordDueToServer(record, "Annotation failed due to server error", isoformOverridesSource);
-            }
-            else if (!summaryStatistics.isFailedAnnotatedRecord(annotatedRecord, record, isoformOverridesSource)) {
+                // logged above when we verify GN response
+                // but still logging for error report
+                summaryStatistics.addFailedAnnotatedRecordDueToServer(record, "Genome Nexus failed to annotate", isoformOverridesSource);
+            } else {
                 annotatedRecord = convertResponseToAnnotatedRecord(gnResponseVariantKeyMap.get(genomicLocationKey), record, replace);
+                if (summaryStatistics.isFailedAnnotatedRecord(annotatedRecord, record, isoformOverridesSource)) {
+                    // Log case where annotation comes back from Genome Nexus but still invalid (e.g null variant classification)
+                    LOG.warn("Annotated record is invalid for variant " + gnResponseVariantKeyMap.get(genomicLocationKey).getVariant());
+                }
             }
             annotatedRecords.add(annotatedRecord);
         }
         return annotatedRecords;
     }
+
+    private List<List<GenomicLocation>> sortAndPartitionMutationRecordsListForPOST(List<GenomicLocation> genomicLocations, Integer postIntervalSize) {
+        // sort
+        List<GenomicLocation> sortedGenomicLocations = new ArrayList<>(genomicLocations);
+        Collections.sort(sortedGenomicLocations, GENOMIC_LOCATION_COMPARATOR);
+        // partition
+        int start = 0;
+        int end = postIntervalSize;
+        List<List<GenomicLocation>> genomicLocationPartitionedLists = new ArrayList<>();
+        while(end <= sortedGenomicLocations.size()) {
```

Comment:
here we make all chunks have the same number of variants (aside from the remainder in the final chunk).  We discussed that this may lead to variable performance .. especially when one chunk spans too many vep cache boundries, requiring many context shifts for the VEP process. We should consider making variable sized chunks so that not too many cache boundries are crossed in a single request.